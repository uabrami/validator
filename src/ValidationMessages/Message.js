import React from "react";

const Message = ({ message }) =>{
  const isDecorated = message.isCompleted ? "line-through" : "";
 return (
  <li style={{textDecoration: isDecorated}}>
    {message.text}
  </li>
 );
};

export default Message;
