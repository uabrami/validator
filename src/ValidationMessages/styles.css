
.messagesContainer {
  display: flex;
  flex-direction: column;
  width: 100%;
}

ul {
  list-style: none;
  padding: 0;
}

ul li::before {
  content: "\2022";
  color: #F07100;
  margin-right: 0.3rem;
  text-decoration: "";
}

ul li {
  padding: 0.5rem 0;
}