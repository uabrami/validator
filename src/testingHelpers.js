import { render } from "@testing-library/react";

export const checkBasicRendering = (component) => {
  it("Expect to render", () => {
    const { container } = render(component);
    expect(container.firstChild).toBeDefined();
  });
};

export const checkRequiredProps = (component) => {
  it("Expect to have all required propTypes", () => {
    const spy = jest.spyOn(global.console, "error");
    render(component);
    expect(spy).not.toHaveBeenCalled();
    spy.mockClear();
  });
};
