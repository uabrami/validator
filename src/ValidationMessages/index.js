import React from "react";
import "./styles.css";

const Message = ({ message }) => <li>{message.text}</li>;

const ValidationMessages = ({messages}) => (
  <div className="Container">
    <div className="row">
      <div className="column">
        <ul>
          {messages.map((message, i) => (
            <Message
              key={i}
              message={message}
            />
          ))}
        </ul>
      </div>
    </div>
  </div>
);


export default ValidationMessages;
