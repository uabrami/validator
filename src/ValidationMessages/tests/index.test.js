import React from "react";
import { render } from "@testing-library/react";

import ValidationMessages from "../index";

import { checkBasicRendering, checkRequiredProps } from "../../testingHelpers";

const testProps = {
  messages: [
    {
      text: "8-72 Characters",
      isCompleted: false,
    },
  ],
};

const Component = <ValidationMessages {...testProps} />;

describe("<ValidationMessages />", () => {
  it("Should render and match the snapshot", () => {
    const {
      container: { firstChild },
    } = render(Component);
    expect(firstChild).toMatchSnapshot();
  });

  checkBasicRendering(Component);
  checkRequiredProps(Component);
});
