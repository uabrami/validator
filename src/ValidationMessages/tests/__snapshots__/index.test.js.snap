// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ValidationMessages /> Should render and match the snapshot 1`] = `
<div
  class="messagesContainer"
>
  <ul>
    <li>
      <span>
        8-72 Characters
      </span>
    </li>
  </ul>
</div>
`;
